assets/settings/default.json

```diff
@@ -142,6 +142,8 @@
   // Whether to display inline and alongside documentation for items in the
   // completions menu
   "show_completion_documentation": true,
+  // Whether to show squiggle underlines for diagnostics in the editor
+  "show_squiggly_underlines": true,
```

Comment:
We’ll need a better name for this setting.

Reply:
Perhaps under `diagnostics` too?
https://github.com/zed-industries/zed/blob/76bf086f94c4ca9651f7e1c0f964af51fe69a66b/assets/settings/default.json#L576-L580